### ISSUE TITLE
[Accessibility]: Global `Menu`: Focus isn't correctly captured when menus are opened

### ISSUE THREAD

**Issue by umpox** [sourcegraph/sourcegraph#36350]
### Audit type

Keyboard navigation

### Problem description

We're having issues when opening menus where it isn't obvious where the focus is, and requires additional steps to select certain items.

For example, see how it isn't immediately obvious in our Menu component from Storybook:
![InaccessibleMenu](https://user-images.githubusercontent.com/9516420/171389682-fa4ab401-6f0b-4063-a005-8a01a6c4d151.gif)

### Expected behavior

When opening a menu with a keyboard, the first item should always be focused.

See this example: https://w3c.github.io/aria-practices/examples/menu-button/menu-button-links.html
![AccessibleMenu](https://user-images.githubusercontent.com/9516420/171389503-f1c0c5df-a003-4f29-b3ba-dcb22e21d13d.gif)

### Additional details

Related issues:
- https://github.com/sourcegraph/sourcegraph/issues/35160
- https://github.com/sourcegraph/sourcegraph/issues/35119
- https://github.com/sourcegraph/sourcegraph/issues/35113

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by umpox** [comment]
Re-opening this.

We have had to revert due to performance issues when using this change with our rendering logic (https://github.com/sourcegraph/sourcegraph/pull/38298)

Will keep this open and aim to address this problem further.

**Comment by felixfbecker** [comment]
I think what's meant here is that the first item is _selected_, the focus is on the menu popup itself I think.
